### PR TITLE
consensus/pbft: log, comments and other slight refactoring

### PR DESCRIPTION
**Pull request by alanchchen**

### CODE REVIEW

File: consensus/pbft/backends/simple/backend.go [hunk 1]
```diff
@@ -121,15 +121,15 @@ func (sb *simpleBackend) Broadcast(payload []byte) error {
...
 	block := &types.Block{}
 	block, ok := proposal.(*types.Block)
 	if !ok {
-		sb.logger.Error("Failed to commit proposal since RequestContext cannot cast to *types.Block")
-		return errCastingRequest
+		sb.logger.Error("Invalid proposal")
+		return errInvalidProposal
 	}
+
+	sb.logger.Info("Committed", "address", sb.Address().Hex(), "hash", proposal.Hash().Hex(), "number", proposal.Number().Uint64())
```
Comment (markya0616) [inline consensus/pbft/backends/simple/backend.go]: sb.Address().Hex() -> sb.Address()
proposal.Hash().Hex() -> proposal.Hash()

File: consensus/pbft/core/handler.go [hunk 2]
```diff
@@ -83,41 +82,46 @@ func (c *core) handleEvents() {
...
 	}
 }
 
+// sendEvent sends internal events
 func (c *core) sendEvent(ev interface{}) {
 	c.backend.EventMux().Post(ev)
 }
 
 func (c *core) handleMsg(payload []byte) error {
 	logger := c.logger.New("address", c.address.Hex())
```
Comment (markya0616) [inline consensus/pbft/core/handler.go]: c.address.Hex() -> c.address

File: consensus/pbft/core/handler.go [hunk 3]
```diff
@@ -83,41 +82,46 @@ func (c *core) handleEvents() {
...
 	}
 
-	return c.handle(msg, src)
+	return c.handleCheckedMsg(msg, src)
 }
 
-func (c *core) handle(msg *message, src pbft.Validator) error {
-	logger := c.logger.New("address", c.address.Hex(), "from", src.Address().Hex())
+func (c *core) handleCheckedMsg(msg *message, src pbft.Validator) error {
+	logger := c.logger.New("address", c.address.Hex(), "from", src)
```
Comment (markya0616) [inline consensus/pbft/core/handler.go]: c.address.Hex() -> c.address

File: consensus/pbft/backend.go [hunk 4]
```diff
@@ -21,50 +21,51 @@ import (
...
-	// Sign is to sign the data
+	// Sign signs input data with the backend's private key
 	Sign([]byte) ([]byte, error)
 
-	// CheckSignature is to verify the signature is signed from given peer
+	// CheckSignature verifies the signature by checking if it's signed by given peer
 	CheckSignature(data []byte, addr common.Address, sig []byte) error
 
 	// CheckValidatorSignature verifies if the data is signed by one of the validators
+	// If the verification success, the signer's address is returned, otherwise
```
Comment (yutelin) [inline consensus/pbft/backend.go]: succeeds

File: consensus/pbft/backends/simple/backend.go [hunk 5]
```diff
@@ -27,18 +27,17 @@ import (
...
 	"github.com/ethereum/go-ethereum/log"
-	"github.com/ethereum/go-ethereum/rlp"
 )
 
 const (
 	extraVanity = 32 // Fixed number of extra-data prefix bytes reserved for signer vanity
 	extraSeal   = 65 // Fixed number of extra-data suffix bytes reserved for signer seal
 )
 
+// New creates a an Ethereum backend for PBFT core engine.
```
Comment (yutelin) [inline consensus/pbft/backends/simple/backend.go]: remove `a`

File: consensus/pbft/backends/simple/backend.go (hunk 1 above)
Comment (yutelin) [inline consensus/pbft/backends/simple/backend.go]: @markya0616  printed binary is not readable right?

File: consensus/pbft/core/handler.go [hunk 6]
```diff
@@ -83,41 +82,46 @@ func (c *core) handleEvents() {
...
 
+	// Store message if its a future message
```
Comment (yutelin) [inline consensus/pbft/core/handler.go]: `it is`

File: consensus/pbft/core/final_committed.go [hunk 7]
```diff
@@ -49,16 +49,15 @@ func (c *core) handleFinalCommitted(ev pbft.FinalCommittedEvent, p pbft.Validato
...
+		// We build a stable checkpoint every 'CheckPointPeriod' proposal
 		if new(big.Int).Mod(c.current.Sequence(), big.NewInt(int64(c.config.CheckPointPeriod))).Int64() == 0 {
 			go c.sendEvent(buildCheckpointEvent{})
 		}
 
 		c.lastProposer = ev.Proposer
 		c.startNewRound(&pbft.View{
 			Sequence: new(big.Int).Add(ev.Proposal.Number(), common.Big1),
-			Round:    common.Big0,
+			Round:    new(big.Int).Set(common.Big0),
```
Comment (alanchchen) [inline consensus/pbft/core/final_committed.go]: Constants under common isn't needed.

File: consensus/pbft/backends/simple/backend.go [hunk 8]
```diff
@@ -163,7 +163,7 @@ func (sb *simpleBackend) Verify(proposal pbft.Proposal) error {
 	block, ok := proposal.(*types.Block)
 	if !ok {
 		sb.logger.Error("Invalid proposal, %v", proposal)
```
Comment (alanchchen) [inline consensus/pbft/backends/simple/backend.go]: remove  `, %v, proposal`

File: consensus/pbft/backends/simple/backend.go (hunk 1 above)
Comment (alanchchen) [inline consensus/pbft/backends/simple/backend.go]: In core/types.go L159

File: consensus/pbft/backends/simple/engine.go [hunk 9]
```diff
@@ -42,13 +42,16 @@ const (
 )
 
 var (
+	// errInvalidProposal is returned when a prposal is malformed.
+	errInvalidProposal = errors.New("invalid proposal")
+	// ErrInvalidSignature is returned when given signature is not signed by given
```
Comment (tailingchen) [inline consensus/pbft/backends/simple/engine.go]: Should be `errInvalidSignature`

**Pull request merged**